location/detector.go

```diff
@@ -1,44 +1,31 @@
 package location
 
 import (
-	"errors"
-	"github.com/oschwald/geoip2-golang"
-	"net"
+	"github.com/mysterium/node/ip"
 )
 
 type detector struct {
-	databasePath string
+	ipResolver ip.Resolver
+	locationResolver Resolver
 }
 
-// NewDetector returns Detector which uses country database
-func NewDetector(databasePath string) *detector {
+func NewLocationDetector(ipResolver ip.Resolver, databasePath string) *detector {
 	return &detector{
-		databasePath: databasePath,
+		ipResolver: ipResolver,
+		locationResolver: NewResolver(databasePath),
 	}
 }
 
-// DetectCountry maps given ip to country
-func (d *detector) DetectCountry(ip string) (string, error) {
-	db, err := geoip2.Open(d.databasePath)
+func (d *detector) DetectCountry() (string, error) {
```

Comment:
Missing doc comments. `bin/lint_git` can show more warnings, just be aware that not all of them are related to your changes.